ASoC: SOF: imx: drop usage of the imx8_*_clocks API

LaurentiuM1234 opened the pull request:
Currently, the driver has to keep track of all the clocks it uses via an array of "struct clk_bulk_data", which doesn't scale well and is unnecessary. As such, replace the usage of the imx8_*_clocks with "devm_clk_bulk_get_all()" and friends.

LaurentiuM1234 committed:
ASoC: SOF: imx: drop usage of the imx8_*_clocks API

Currently, the driver has to keep track of all the clocks
it uses via an array of "struct clk_bulk_data", which doesn't
scale well and is unnecessary. As such, replace the usage of
the imx8_*_clocks with "devm_clk_bulk_get_all()" and friends.

Signed-off-by: Laurentiu Mihalcea <laurentiu.mihalcea@nxp.com>

dbaluta commented:
Looks good to me @LaurentiuM1234. We can now remove the implementation of `imx8_parse_clocks` from imx-common.c also.

LaurentiuM1234 committed:
ASoC: SOF: imx: remove imx8_*_clocks API

Since the i.MX drivers no longer use the imx8_*_clocks API
this can be removed.

Signed-off-by: Laurentiu Mihalcea <laurentiu.mihalcea@nxp.com>

LaurentiuM1234 commented:
**V2 changes**

1. Added patch that removes the `imx8_*_clocks` API.

dbaluta commented:
@lgirdwood  this is good to merge.
